src/main/java/core/basesyntax/operations/PurchaseOperation.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void apply(Storage storage, FruitTransaction transaction) {

```

Comment:
The method signature of `apply` does not follow the encapsulation principle. The `Storage` should not be passed directly to operation handlers. Instead, consider using a service that abstracts the interaction with the `Storage`.